### DISCUSSION TITLE
Check for duplicate tracks

### DISCUSSION THREAD

**Discussion by jhurlbut** [okio-ai/nendo-platform#10] [Ideas]
It would be nice if the database would check for duplicates if you import the same file twice instead of adding a new duplicate track

**Comment by faradox** [comment]
Thanks for this suggestion! Nendo Core already provides support for duplicate track checking, we will hook this feature into the platform asap.

**Comment by faradox** [comment]
See https://github.com/okio-ai/nendo-platform/issues/25

**Comment by jhurlbut** [comment]
Cool! It's not working relialably for me. The first time I uploaded a duplicate it gave me the upload complete popup but did not add the duplicate as expected. I tried adding it again and it did add a duplicate. Then I deleted both tracks from the library and uploaded a third time and got a upload failed popup but it did actually import the track.

Some other comments:
It would be nice to see a confirmation dialog like "It looks like this track is already in the database"
maybe a checkbox in the import popup "skip duplicate tracks" for multi file upload
It would be nice if the upload finished popup said something like "The track wasn't added because it's a duplicate" instead of "Upload Complete" or "upload failed"

**Comment by faradox** [comment]
I couldn't reproduce your issue. When I upload the same file with `skip_duplicate=true`, it does not add the track a second time, no matter how often I try (note: _same file_, not the same track but two different files, that kind of duplication detection can get very technically complex and is not going to be supported any time soon).

As for your other comments: Excellent ideas, thanks for your input! Will include some improvements in the next update.
